### PR TITLE
[batch] Added CloudSQL capabilities for batch

### PR DESCRIPTION
**Pull request by jigold**
- Only batch for now -- will add for pipeline and ci later
 - This should fail until the kubernetes secret is added
- Requires a password `CLOUD_SQL_PASSWORD` to run the tests locally
    (not sure what the best way to distribute this is)
- Requires downloading the `cloud_sql_proxy` binary to run the tests locally

### CODE REVIEW

File: batch/test-locally.sh [hunk 1]
```diff
@@ -9,10 +9,30 @@ cleanup() {
...
+    proxy_pid=$!
+    CLOUD_SQL_HOST='127.0.0.1'
+else
+    CLOUD_SQL_HOST='10.80.0.3'
+    CLOUD_SQL_PORT='3307'
+fi
+
+while [[ -z $CLOUD_SQL_DB ]]; do
+    CLOUD_SQL_DB="test-${uuid:-$(../generate-uid.sh)}"
+    if gcloud sql databases list --uri -i $CLOUD_SQL_INSTANCE | grep $CLOUD_SQL_DB; then
```
Comment (danking) [inline batch/test-locally.sh]: probably should `--quiet` grep

File: batch/Makefile [hunk 2]
```diff
@@ -59,12 +65,27 @@ ifneq ($(PY_CHECK),false)
 test-local: $(PY_CHECKERS)
 endif
 test-local: build/conda-env
-	. ../loadconda && conda activate hail-batch && POD_NAMESPACE='test' BATCH_USE_KUBE_CONFIG=1 ./test-locally.sh
```
Comment (danking) [inline batch/Makefile]: probably should add a line to this makefile:
```
ifndef CLOUD_SQL_PASSWORD
$(error CLOUD_SQL_PASSWORD is not set)
endif
```

Is the password in the usual place?

Comment (jigold) [reply batch/Makefile]: No. I put it in my environment. Not sure what the best strategy is for this.

**Comment by jigold** [comment]
Depends on #5679

File: cloud-sql.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+install-cloud-sql-proxy:
+	test -f cloud_sql_proxy || \
+		curl -o cloud_sql_proxy https://dl.google.com/cloudsql/cloud_sql_proxy.darwin.amd64 && \
+		chmod +x cloud_sql_proxy
+
+local-cloud-sql-config:
+	mkdir -p batch-secrets
+	kubectl get secret batch-test-cloud-sql-config -n batch-pods -o "jsonpath={.data.batch-test-cloud-sql-config\.json}" | base64 --decode > batch-secrets/batch-test-cloud-sql-config.json
+	jq '.host = "127.0.0.1" | .port = $(CLOUD_SQL_PORT)' batch-secrets/batch-test-cloud-sql-config.json > batch-secrets/tmp.batch-test-cloud-sql-config.json
+	mv batch-secrets/tmp.batch-test-cloud-sql-config.json batch-secrets/batch-test-cloud-sql-config.json
```
Comment (danking) [inline cloud-sql.mk]: We can do it in one full pipe:
```
	kubectl get secret \
	    batch-test-cloud-sql-config -n batch-pods \
	    -o "jsonpath={.data.batch-test-cloud-sql-config\.json}" \
	  | base64 --decode \
	  | jq '.host = "127.0.0.1" | .port = $(CLOUD_SQL_PORT)' \
	    > batch-secrets/batch-test-cloud-sql-config.json
```

And I think I'd make this a non-PHONY target. If the secret changes, that will break folks, but I think that happening rarely is better than having to run `kubectl` every time we run the tests, right?

**Comment by jigold** [comment]
It's finally passing the tests!

**Comment by danking** [comment]
@jigold can you check batch in an hour and/or tomorrow morning to make sure its all good after deploy too?

**Pull request merged**